e2e/app.test.js

```diff
@@ -0,0 +1,274 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
```

Comment:
I do not understand why this test exist? It launches a browser and runs tests just like the puppeteer E2E:alone test does.
IMO we should have 3 tests.
1) functionality tests (you already have build those)
2) E2E test (1 happy path test that shows that flow works - already built those)
3) component tests (these does not have to open a browser in order to run)

The way this is implemented IMO is not correct as it takes way too much time to run (open browser, wait, run, etc etc) and it should be able to run also in CI (which this kind of tests are not able i think)

Reply:
Well i think that it does not have to be so drastic... the additional tests added do not add more than 30 seconds or similar so even if it adds minutes to the process is something we can deal with more taking into account that the gain is significant in terms of consistency and quality.

Then I agree with your three suggestions but the purpose of this PR is to extend and cover all the critical path of the transfer feature and the complex logic around the receiver field that needs to work perfectly and this tests validate that.

The reason for including the tests on this jest file is because we can combine puppeter tools with powerful jest assertions which i think it's the best of two worlds.

Finally , the next step is to include these tests to the CI so we can run these tests in every deploy of master branch: #192

Reply:
IMO all test should not take more than 20 seconds. In Substrate connect, covering 2 major projects with approx 80-90% coverage of unit tests takes 20 seconds. 
In addition i do not understand how exactly these tests will run in CI

Reply:
Ok fair enough ... 

The goal for the tests are described in the description of this PR:

Test 1: Should provide native source address type and should show native address in receiver field.
Test 2: Should provide companion source address type and should show companion address in receiver field.
Test 2: Should provide generic address type and should show two options, choose native and get native receiver field.
Test 4: Should execute end to end testing of transaction. This includes the following checks:

Reply:
I know. I guess what i am trying to say is that these tests can run without even initiating a browser as far as I know

Reply:
We shouldn't be adding any more E2E tests for now, but the ones that are here can stay. The focus should be more on unit testing, until we get some major features (like a new tab). 

Running it on the CI is a priority (at least after merging to master and deploying).

> I guess what i am trying to say is that these tests can run without even initiating a browser as far as I know

What's the alternative? Does pupetteer have some built-in browser?

Reply:
> We shouldn't be adding any more E2E tests for now, but the ones that are here can stay. 

I agree. As soon as the work is done - there is no need for removing or anything.

> Running it on the CI is a priority

Im not sure how one can run GUI tests in Github CI - I do not say that it cannot happen - I just do not know how (never done it)

> Does puppetteer have some built-in browser

Puppeteer's browsers can be downloaded along with it (meaning you can add chromium and/or firefox with the installation of the package and run the tests with the downloaded browsers (cross-browsing testing). 
The configuration I did, does not include any browser and it is assumed that chrome is installed with it.

On top of that, puppeteer can run headless browsers (without the need of any GUI) and that is my point above. Lets run the E2E or component tests, but not with full browsers and without such a waiting time. 

I would suggest - since new components implementation start - to pause unit (component tests for now) and implement them whenever a new component is created (e.g. build a new dropdown -> write also unit/component tests for it on the way).

Reply:
@wirednkod the browser setup is something i plan to work on #192, this PR only includes the new tests described before.